README.md

```diff
@@ -552,120 +569,153 @@ times a second. You can test the following properties:
+like [cheerio](https://npmjs.com/package/cheerio) for more sophisticated
+resource discovery.
+- **Q: Why did simplecrawler complete without fetching any resources?**
+A: When this happens, it is usually because the initial request was redirected
+to a different domain that wasn't in the `domainWhitelist`.
+- **Q: What does it mean that events are asynchronous?**
+A: One of the core concepts of node.js is its asynchronous nature. I/O
+operations (like network requests) take place outside of the main thread (which
+is where your code is executed). This is what makes node fast, the fact that it
+can continue executing code while there are multiple HTTP requests in flight,
+for example. But to be able to get back the result of the HTTP request, we need
+to register a function that will be called when the result is ready. This is the
+same concept as with AJAX requests in the browser.
+- **Q: Promises are nice, can I use them with simplecrawler?**
+A: No, not really. Promises are meant as a replacement for callbacks, but
+simplecrawler is event driven, not callback driven. Using callbacks to any
+greater extent in simplecrawler wouldn't make much sense, since you normally
+need to react more than once to what happens in simplecrawler.
+
 ## Contributors
 
 I'd like to extend sincere thanks to:
 
-*   [Nick Crohn](https://github.com/ncrohn) for the HTTP Basic auth support, and
-    initial cookie support.
-*   [Mike Moulton](https://github.com/mmoulton) for
-    [fixing a bug in the URL discovery mechanism]
-    (https://github.com/cgiffard/node-simplecrawler/pull/3), as well as
-    [adding the `discoverycomplete` event]
-    (https://github.com/cgiffard/node-simplecrawler/pull/10),
-*   [Mike Iannacone](https://github.com/mikeiannacone) for correcting a keyword
-    naming collision with node 0.8's EventEmitter.
-*   [Greg Molnar](https://github.com/gregmolnar) for
-    [adding a querystring-free path parameter to parsed URL objects.]
-    (https://github.com/cgiffard/node-simplecrawler/pull/31)
-*   [Breck Yunits](https://github.com/breck7) for contributing a useful code
-    sample demonstrating using simplecrawler for caching a website to disk!
-*   [Luke Plaster](https://github.com/notatestuser) for enabling protocol-agnostic
-    link discovery
-*   [Zeus](https://github.com/distracteddev) for fixing a bug where [default port
-    info was wrongly specified in requests]
-    (https://github.com/cgiffard/node-simplecrawler/pull/40)
-    and for fixing the missing request timeout handling!
-*   [Graham Hutchinson](https://github.com/ghhutch) for adding
-    querystring-stripping option
-*   [Jellyfrog](https://github.com/jellyfrog) for assisting in diagnosing some
-    nasty EventEmitter issues.
-*   [Brian Moeskau](https://github.com/bmoeskau) for helping to fix the confusing
-    'async' events API, and providing invaluable feedback.
+* [Nick Crohn](https://github.com/ncrohn) for the HTTP Basic auth support, and
```

Comment:
No worries :) I think maybe this section could be turned into some contributor guidelines, encouraging people to submit issues and PR's, and thanking all of those who've done it (not by name, but just like "thanks to all of those who've contributed"). Could be a link to the contributor graph as well!